### DISCUSSION TITLE
Micronaut with Kotlin Class having @Singleton annotation not instantiated

### DISCUSSION THREAD

**Discussion by nanil4568** [micronaut-projects/micronaut-core#5254] [Q&A] [answered]
I have an Micronaut application with modules written in Java and Kotlin.
Java module(Controller logic) has an interface, which is implemented by Kotlin Class with @Singleton annotation specified. But at runtime the Kotlin implementation class is not getting instantiated and is failing with the below exception.

"Internal Server Error: Failed to inject value for parameter [<Interface name>] of class. No bean of type [<Interface class path>] exists..."

The same works if the implementation class is also written in Java.
Is this a known issue? Please let me know if I'm missing some configuration to make it work

**Comment by graemerocher** [comment, marked as answer]
Provide an example and maybe we can help

Answer provided at https://stackoverflow.com/questions/67172596/micronaut-with-kotlin-class-having-singleton-annotation-not-instantiated/67174786#67174786

**Reply by nanil4568** [reply]
Attached the project having the sample controller(DemoController.java), interface(Message.java) and Kotlin implementation class(MessageImpl.kt)
Issue can be reproduced by starting the server and then invoking the endpoint "http://localhost:8080/hello"


[demo.zip](https://github.com/micronaut-projects/micronaut-core/files/6317544/demo.zip)

**Reply by nanil4568** [reply]
@graemerocher can you please take a look and provide inputs

**Reply by ilopmar** [reply]
@nanil4568 I've replied in your Stackoverflow question: https://stackoverflow.com/questions/67172596/micronaut-with-kotlin-class-having-singleton-annotation-not-instantiated/67174786#67174786

**Reply by yanalapraneetha** [reply]
@ilopmar we have applied the mentioned dependencies from the stack overflow build.gradle, after that injection of kotlin object from java class is working, but somehow we are still facing the same bean initialization exception when we try to inject a singleton kotlin class in another kotlin class. It is working as expected and object is getting injected in demo project but in our actual project this issue is seen, we have compared all the dependencies between 2 projects we have noticed that in demo project under Gradle:kaptGeneratedClasses there are some classes where as in our actual project kaptGeneratedClasses folder is empty. I am suspecting this could be related to the issue we are seeing. 
Can you please provide your inputs? 

Exception -
Internal Server Error: Failed to inject value for parameter [] of class Message: No bean of type class exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor)
